helios.js

```diff
@@ -28652,6 +28709,98 @@ class InlineDatum extends Datum {
 	}
 }
 
+/**
+ * Encoding for Transcation Matadatum
+ */
+export class MetadataString extends CborData {
+	/** @type {number[]} */
+	#bytes;
+
+	/**
+	 * @param {number[]} bytes
+	 */
+	constructor(bytes) {
+		super();
+		this.#bytes = bytes;
+	}
+
+	/**
+	 * Applies utf-8 encoding
+	 * @param {string} s
+	 * @returns {MetadataString}
+	 */
+	static fromString(s) {
+		let bytes = stringToBytes(s);
+
+		return new MetadataString(bytes);
+	}
+
+	get bytes() {
+		return this.#bytes.slice();
+	}
+
+	/**
+	 * @returns {string}
+	 */
+	toHex() {
+		return bytesToHex(this.#bytes);
+	}
+
+	toString() {
+		return `#${this.toHex()}`;
+	}
+
+	/**
+	 * @returns {number[]}
+	 */
+	toCbor() {
+		// If the data is >64 bytes then split into List
+
+		if (this.#bytes.length > 64) {
+			const chunks = [];
+			for (let i = 0; i < this.#bytes.length; i = i + 64) {
+				chunks.push(CborData.encodeUtf8(this.#bytes.slice(i, i + 64)));
+			}
+			return CborData.encodeDefList(chunks);
+		}
+
+		return CborData.encodeUtf8(this.#bytes);
+	}
+}
+
+class TxMetadata {
+	/**@type {Record<number, number[]>} */
```

Comment:
I haven't used the `Record` container type anywhere in helios.js, perhaps we could use `Object` instead?

Reply:
Keeping my other comments in mind: the value type could be `Metadata`